### PR TITLE
Hercules' Pickaxe supports 1.17 deepslate ores and copper

### LINKED ISSUE #3251: Hercules Pickaxe not turning Copper Ore, or Iron/Gold Deepslate Ore into dust.
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When attempting to mine Iron Deepslate Ore or Gold Deepslate Ore with the Hercules' Pickaxe, instead of turning into dust, it drops normal items; Raw Iron and Raw Gold. Also, when mining Copper Ore or Copper Deepslate Ore with the Hercules' Pickaxe, and the same occurs.

### 📑 Reproduction Steps

1) Find/Place one of the three ores mentioned above (Iron/Gold/Copper Deepslate Ores, or just regular Copper Ore).
2) Obtain, in some way, the Hercules' Pickaxe
3) Mine the Ore.

### 💡 Expected Behavior

I was expecting for the Iron Deepslate Ore to drop Iron Dust, the Gold Deepslate Ore to drop Gold Dust, and either variant of the Copper Ore to drop Copper Dust.

### 📷 Screenshots / Videos

_No response_

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.17.x

### ⭐ Slimefun version

Slimefun DEV - 968 (git 79c6c139)

### 🧭 Other plugins

_No response_

### PR DESCRIPTION
**Pull request by ybw0014**
## Description

The Hercules' Pickaxe should function like Ore Crusher, so the drops of mining deepslate ores or copper ores should be their corresponding dusts.

## Proposed changes

Add support for Hercules' Pickaxe with 1.17 deepslate ores and copper ores.

## Related Issues (if applicable)

Resolves #3251

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java [hunk 1]
```diff
@@ -25,16 +27,36 @@ public HerculesPickaxe(ItemGroup itemGroup, SlimefunItemStack item, RecipeType r
...
+                        case DEEPSLATE_IRON_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.IRON_DUST, 2));
+                            break;
+                        case DEEPSLATE_GOLD_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.GOLD_DUST, 2));
+                            break;
+                        case COPPER_ORE:
+                        case DEEPSLATE_COPPER_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.COPPER_DUST, 2));
+                            break;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:44]: Missing `default:` statement

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java (hunk 1 above)
Comment (ybw0014) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:44]: this section is about 1.17 ores, if the block type does not match here, it should then go to next switch

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:44]: This switch is still missing a `default: break;` statement though.
Having no 'default' statement usually implies that there are no unhandled possible values.
But this `switch` does not cover all values of the `Material` enum, hence why it definitely needs a `default` statement from a semantical view.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java [hunk 2]
```diff
@@ -25,16 +27,36 @@ public HerculesPickaxe(ItemGroup itemGroup, SlimefunItemStack item, RecipeType r
...
+                    case IRON_ORE:
+                        drops.add(new CustomItemStack(SlimefunItems.IRON_DUST, 2));
+                        break;
+                    case GOLD_ORE:
+                        drops.add(new CustomItemStack(SlimefunItems.GOLD_DUST, 2));
+                        break;
+                    default:
+                        for (ItemStack drop : e.getBlock().getDrops(tool)) {
+                            drops.add(new CustomItemStack(drop, drop.getAmount() * 2));
+                        }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:58]: Also missing `break;`

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM I suppose.

Though the more I look at it, the more I start to question why this item even exists...

**Review by svr333 — APPROVED** [review]
Looks good, maybe regarding @TheBusyBiscuit's comment, we can deprecate the item

**Pull request merged**